File: Data Structures & Algorithms/binary-search/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {
        let low = 0, high = nums.length - 1;
        while(low<=high){
            const mid = low  + Math.floor((high - low)/2);
            if(target === nums[mid]) return mid;
            if(target < nums[mid]){
                high = mid - 1
            }
            if(target > nums[mid]){
                low = mid + 1
            }
        }
        return -1
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-0.py
import math
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        low, high, ans = 1, max(piles), max(piles)
        while low <= high:
            mid = (low + high) // 2
            if self.isFeasible(piles, mid, h):
                ans = min(ans, mid)
                high = mid - 1
            else:
                low = mid + 1
        return ans

    def isFeasible(self, piles, k, h) -> bool:
        sum = 0
        for i in piles:
            sum+=math.ceil(i / k)
        return sum <= h

        